src/Client.php

```diff
@@ -329,6 +331,19 @@ public function selectDatabase(string $databaseName, array $options = [])
         return new Database($this->manager, $databaseName, $options);
     }
 
+    public function setLogger(LoggerInterface $logger): void
+    {
+        /* TODO: lazily initialize a PsrLogAdapter for this client and register
+         * it with PHPC. Consider what happens if multiple clients instances in
+         * an app register the same PSR logger. That logger will likely receive
+         * duplicate messages, since PHPC's own checks would only prevent the
+         * same PsrLogAdapter instance from being registered multiple times.
+         *
+         * Perhaps we can work around this by making PsrLogAdapter a singleton
+         * or utilize a static map of PSR loggers.
```

Comment:
A static map of PSR loggers may be sensible - I don't think a singleton is the right choice here. Technically, we'd want to have one instance of `PsrLogAdapter` per `LoggerInterface` at most, so not quite a singleton.

Reply:
The current implementation is one `PsrLogAdapter` per `LoggerInterface`. The proposition would be to have a single `PsrLogAdapter` (singleton) sending to all `LoggerInterface` that it contains. The adapter would have methods to add and remove `LoggerInterface` objects.

Reply:
Sorry, I wasn't actually suggesting that we make PsrLogAdapter a singleton. A static property is what I had in mind, as that would allow all PsrLogAdapter instances (at most one per Client) to track which loggers have been globally registered.

```mermaid
graph TD
    A[Client1] --> B[PsrLogAdapter1]
    B --> C[PsrLogger]
    D[Client2] --> E[PsrLogAdapter2]
    E --> C
```

In this example, the static map shared by all PsrLogAdapter instances would ensure that PHPLIB never registers the PsrLogAdapter a second time.

Trying to think of how this could be problematic:

 1. Construct Client1 and assign PsrLogger1 to it. PsrLogAdapter1 is created and registered with PHPC.
 2. Construct Client2 and assign PsrLogger1 to it. PsrLogAdapter2 is created but not registered, since it knows another PsrLogAdapter instance has already been registered for PsrLogger1.
 3. Destroy Client1 and allow PsrAdapter1 to be unregistered.

In that case, Client2's logging no longer functions because PsrLogAdapter2 was never registered. I'll give this some more thought.

Reply:
Closing the loop here. PsrLogAdapter is now a singleton and collects one or more PsrLoggers. The PsrLogAdapter registered itself when the first PsrLogger is added and unregistered itself when the last logger is removed.